tests/integration/conftest.py

```diff
@@ -247,33 +260,40 @@ async def app_openstack_runner(
     no_proxy: str,
     openstack_clouds_yaml: str,
     openstack_flavor: str,
+    existing_app: Optional[str],
 ) -> AsyncIterator[Application]:
     """Application launching VMs and no runners."""
-    application = await deploy_github_runner_charm(
-        model=model,
-        charm_file=charm_file,
-        app_name=app_name,
-        path=path,
-        token=token,
-        runner_storage="juju-storage",
-        http_proxy=http_proxy,
-        https_proxy=https_proxy,
-        no_proxy=no_proxy,
-        reconcile_interval=60,
-        constraints={
-            "root-disk": 20 * 1024,
-            "cores": 3,
-            "mem": 12 * 1024,
-            "virt-type": "virtual-machine",
-        },
-        config={
-            OPENSTACK_CLOUDS_YAML_CONFIG_NAME: openstack_clouds_yaml,
-            # this is set by microstack sunbeam, see scripts/setup-microstack.sh
-            OPENSTACK_NETWORK_CONFIG_NAME: "demo-network",
-            OPENSTACK_FLAVOR_CONFIG_NAME: openstack_flavor,
-        },
-        wait_idle=False,
-    )
+    if existing_app:
+        application = model.applications[existing_app]
+    else:
+        application = await deploy_github_runner_charm(
+            model=model,
+            charm_file=charm_file,
+            app_name=app_name,
+            path=path,
+            token=token,
+            runner_storage="juju-storage",
+            http_proxy=http_proxy,
+            https_proxy=https_proxy,
+            no_proxy=no_proxy,
+            reconcile_interval=60,
+            constraints={
+                "root-disk": 50 * 1024,
+                "cores": 4,
+                "mem": 16 * 1024,
+                "arch": "arm64",
+            },
+            config={
+                OPENSTACK_CLOUDS_YAML_CONFIG_NAME: openstack_clouds_yaml,
+                # this is set by microstack sunbeam, see scripts/setup-microstack.sh
+                OPENSTACK_NETWORK_CONFIG_NAME: "demo-network",
```

Comment:
I hardcoded this due to using microstack previously. 
I would think this will need to be different for private endpoints. How did the test work with this hardcoded value.

This would need to be an option and pass via secret, which I can do in a follow-up PR.

Reply:
It seems this app was not used for testing. Which would make sense why the tests passed.

Reply:
Yes, I replaced these config options locally with something appropriate for our staging environment, but I did not want to check this into the source code, so I kept the previous value for the sake of this PR. I figured you would change it in the next PR anyway.